systems/trajectory_optimization/hybrid_dircon.cc

```diff
@@ -66,118 +69,187 @@ HybridDircon<T>::HybridDircon(
     for (int j = 0; j < mode_lengths_[i] - 2; j++) {
       // all timesteps must be equal
       AddLinearConstraint(timestep(mode_start_[i] + j) ==
-                          timestep(mode_start_[i] + j + 1));
+          timestep(mode_start_[i] + j + 1));
     }
 
     // initialize constraint lengths
+    num_kinematic_constraints_wo_skipping_.push_back(
+        constraints_[i]->countConstraintsWithoutSkipping());
     num_kinematic_constraints_.push_back(constraints_[i]->countConstraints());
 
     // initialize decision variables
-    force_vars_.push_back(NewContinuousVariables(constraints_[i]->countConstraints() * num_time_samples[i], "lambda[" + std::to_string(i) + "]"));
-    collocation_force_vars_.push_back(NewContinuousVariables(constraints_[i]->countConstraints() * (num_time_samples[i] - 1), "lambda_c[" + std::to_string(i) + "]"));
-    collocation_slack_vars_.push_back(NewContinuousVariables(constraints_[i]->countConstraints() * (num_time_samples[i] - 1), "v_c[" + std::to_string(i) + "]"));
-    offset_vars_.push_back(NewContinuousVariables(options[i].getNumRelative(), "offset[" + std::to_string(i) + "]"));
+    force_vars_.push_back(NewContinuousVariables(
+        constraints_[i]->countConstraintsWithoutSkipping()
+            * num_time_samples[i], "lambda[" + std::to_string(i) + "]"));
+    collocation_force_vars_.push_back(NewContinuousVariables(
+        constraints_[i]->countConstraintsWithoutSkipping()
+            * (num_time_samples[i] - 1),
+        "lambda_c[" + std::to_string(i) + "]"));
+    collocation_slack_vars_.push_back(NewContinuousVariables(
+        constraints_[i]->countConstraintsWithoutSkipping()
+            * (num_time_samples[i] - 1), "v_c[" + std::to_string(i) + "]"));
+    // slack variables used to scale quaternion norm to 1 in the dynamic constraints.
+    quaternion_slack_vars_.push_back(NewContinuousVariables(
```

Comment:
This isn't very readable. Why not
```
if (is_quaternion) {
// Add variables

}
```

Reply:
I feel they are equally readable, but I will change to 
```
if (is_quaternion) {
// Add variables

}
```